client/ayon_aftereffects/hooks/pre_launch_install_ayon_extension.py

```diff
@@ -0,0 +1,154 @@
+                f"The extension already exists at: {target_path}. "
+                f"Comparing versions.."
+            )
+            if not self._compare_extension_versions(
+                target_path, extension_path
+            ):
+                return
+
+        try:
+            self.log.debug(f"Creating directory: {target_path}")
+            os.makedirs(target_path, exist_ok=True)
+
+            with ZipFile(extension_path, "r") as archive:
+                archive.extractall(path=target_path)
+            self.log.info("Successfully installed AYON extension")
+
+        except OSError as error:
+            self.log.warning(f"OS error has occured: {error}")
+
+        except PermissionError as error:
+            self.log.warning(f"Permissions error has occured: {error}")
+
+        except Exception as error:
+            self.log.warning(f"An unexpected error occured: {error}")
+
+    def _compare_extension_versions(
+        self, target_path: str, extension_path: str
+    ) -> bool:
+        try:
+            # opens the existing extension manifest to get the Version attr
+            with open(f"{target_path}/CSXS/manifest.xml", "rb") as xml_file:
+                installed_version = (
+                    ET.parse(xml_file)
+                    .getroot()
+                    .attrib.get("ExtensionBundleVersion")
+                )
+            self.log.debug(
+                f"Current extension version found: {installed_version}"
+            )
+
+            if not installed_version:
+                self.log.warning(
+                    "Unable to resolve the currently installed extension "
+                    "version. Cancelling.."
+                )
+                return False
+
+            # opens the .zxp manifest to get the Version attribute.
+            with ZipFile(extension_path, "r") as archive:
+                xml_file = archive.open("CSXS/manifest.xml")
+                new_version = (
+                    ET.parse(xml_file)
+                    .getroot()
+                    .attrib.get("ExtensionBundleVersion")
+                )
+                if not new_version:
+                    self.log.warning(
+                        "Unable to resolve the new extension version. "
+                        "Cancelling.."
+                    )
+                self.log.debug(f"New extension version found: {new_version}")
+
+                # compare the two versions, a simple == is enough since
+                # we don't care if the version increments or decrements
+                # if they match nothing happens.
+                if installed_version == new_version:
+                    self.log.info("Versions matched. Cancelling..")
+                    return False
+
+                # remove the existing addon
+                self.log.info(
+                    "Version mismatch found. Removing old extensions.."
+                )
+                rmtree(target_path)
+                return True
```

Comment:
I think we don't need to have all the code under `with open` context manager?
```suggestion
            new_version = (
                ET.parse(xml_file)
                .getroot()
                .attrib.get("ExtensionBundleVersion")
            )
            if not new_version:
                self.log.warning(
                    "Unable to resolve the new extension version. "
                    "Cancelling.."
                )
            self.log.debug(f"New extension version found: {new_version}")

            # compare the two versions, a simple == is enough since
            # we don't care if the version increments or decrements
            # if they match nothing happens.
            if installed_version == new_version:
                self.log.info("Versions matched. Cancelling..")
                return False

            # remove the existing addon
            self.log.info(
                "Version mismatch found. Removing old extensions.."
            )
            rmtree(target_path)
            return True
```

Reply:
Good spot! I won't have a chance to validate this until next week. It looks correct so happy to commit if you've tested, if not I'll test asap and commit

Reply:
Should be resolved now